datajoint/table.py

```diff
@@ -344,23 +348,49 @@ def _delete_cascade(self):
             try:
                 delete_count += self.delete_quick(get_count=True)
             except IntegrityError as error:
-                match = foregn_key_error_regexp.match(error.args[0])
-                assert match is not None, "foreign key parsing error"
+                try:
```

Comment:
`try ... except` should enclose the minimal statement that's expected to raise an exception.

Reply:
@dimitri-yatsenko Good point. I think I can adjust this to remove the exception all together. Let me try to refine this.